### PR TITLE
chore(): upgrade @angular/cli to rc.0

### PR DESCRIPTION
**Pull request by emoralesb05**
## Description

Upgrade @angular/cli dependency to rc.0

https://github.com/angular/angular-cli/blob/master/docs/documentation/stories/rc.0-update.md

#### Test Steps

- [ ] `npm uninstall -g @angular/cli`
- [ ] `npm cache clean`
- [ ] `npm install -g @angular/cli`
- [ ] `npm run reinstall`

#### General Tests for Every PR

- [ ] `ng serve --aot` still works.
- [ ] `npm run lint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build` still works.

##### Screenshots or link to CodePen/Plunker/JSfiddle

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -115,8 +114,8 @@
     "rollup-plugin-node-resolve": "2.0.0",
     "sass-loader": "^4.0.2",
```
Comment (d3viant0ne) [inline package.json]: Is `sass-loader` still used in the project? `node-sass` is used in the gulp scripting but pulling the loader has no effect on the project ( left over from a version override? )

File: src/tsconfig.app.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "noUnusedParameters": false,
+    "noUnusedLocals": false,
+    "allowUnreachableCode": false,
+    "pretty": true
+  },
+  "exclude": [
+    "test.ts",
+    "**/*.spec.*"
+  ]
+}
```
Comment (d3viant0ne) [inline src/tsconfig.app.json]: In rc.0 the app & spec typescript configs extend a base config.

`./src/tsconfig.app.json`
```
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/app",
    "module": "es2015",
    "target": "es5",
    "baseUrl": ".",
    "types": [
      "hammerjs"
    ]
  },
  "exclude": [
    "test.ts",
    "**/*.spec.ts"
  ]
}

```

`./tsconfig.json`
```
{
  "compileOnSave": false,
  "compilerOptions": {
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "noUnusedParameters": false,
    "noUnusedLocals": false,
    "allowUnreachableCode": false,
    "pretty": true,
    "lib": [
      "es2016",
      "dom"
    ],
    "typeRoots": [
      "../node_modules/@types"
    ],
    "paths": {
      "@covalent/*": ["./platform/*"]
    }
  }
}


```

File: src/tsconfig.spec.json [hunk 3]
```diff
@@ -25,7 +25,7 @@
     "allowUnreachableCode": false,
     "pretty": true
   },
-  "exclude": [
+  "include": [
     "**/*.spec.*"
   ]
 }
```
Comment (d3viant0ne) [inline src/tsconfig.spec.json]: _See comment above_

`./src/tsconfig.spec.json`
```
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/spec",
    "module": "commonjs",
    "target": "es5",
    "baseUrl": ".",
    "types": [
      "jasmine",
      "hammerjs",
      "node"
    ]
  },
  "files": [
    "test.ts"
  ],
  "include": [
    "**/*.spec.ts"
  ]
}

```

**Comment by d3viant0ne** [comment]
Also worth noting...You are going to want to disable tslint for `pollyfill.ts` & `test.ts` until they fix the jsdoc formatting & empty blocks

File: package.json (hunk 1 above)
Comment (emoralesb05) [inline package.json]: Removing it 👍  it was there since we started the project haha

**Pull request merged**